File: src-tauri/src/core/cmd/window.rs
use crate::utils::window_manager;

#[tauri::command]
pub async fn toggle_window(window_type: String) {
    window_manager::toggle_window_by_label(&window_type);
}

#[tauri::command]
pub async fn show_window(window_type: String,url: Option<String>) {
    window_manager::show_window_by_label(&window_type,url.as_deref());
}

#[tauri::command]
pub async fn close_window(window_type: String) {
    window_manager::close_window_by_label(&window_type);
}

#[tauri::command]
pub async fn minimize_window(window_type: String) -> bool {
    window_manager::minimize_window_by_label(&window_type)
}

// 主要是用来关闭浮动窗口，避免窗口被缓存
#[tauri::command]
pub async fn destroy_window(window_type: String) -> bool {
    window_manager::destroy_window_by_label(&window_type)
}

File: src/api/modules/window.ts
import { invoke } from "@tauri-apps/api/core";


function toggleWindow(windowType: string) {
    return invoke("toggle_window", { windowType });
}

function showWindow(windowType: string, url?: string) {
    return invoke("show_window", { windowType, url });
}

function minimizeWindow(windowType: string) {
    return invoke("minimize_window", { windowType });
}

function closeWindow(windowType: string) {
    return invoke("close_window", { windowType });
}

function destroyWindow(windowType: string) {
    return invoke("destroy_window", { windowType });
}

export { toggleWindow, showWindow, minimizeWindow, closeWindow, destroyWindow };
